src/main/steps/applicant1/irretrievable-breakdown/post.ts

```diff
@@ -0,0 +1,24 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { AppRequest } from '../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../app/controller/PostController';
+import { Form, FormFields } from '../../../app/form/Form';
+
+@autobind
+export default class IrretrievableBreakdownPostController extends PostController<AnyObject> {
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const form = new Form(<FormFields>this.fields);
+
+    const { saveAndSignOut, ...originalFormData } = form.getParsedBody(req.body);
+    const formData = originalFormData;
+
+    formData.divorceOrDissolution = req.session.userCase.divorceOrDissolution;
+
+    if (req.body.saveAndSignOut || req.body.saveBeforeSessionTimeout) {
+      await this.saveAndSignOut(req, res, formData);
+    } else {
+      await this.saveAndContinue(req, res, form, formData);
+    }
+  }
+}
```

Comment:
I'm not sure this needs a custom post controller, the only difference is this sets the divorceOrDissolution field in the formData, but I can't see why we'd need to do that?

Reply:
discussed and agreed this is the right approach as without this the divorceOrDissolution field is undefined and this causes the conversion to API to fail as it's required in order to convert correctly. Marking as resolved.